### ISSUE TITLE
Rule: use only arrow functions in React component 

### ISSUE THREAD

**Issue by spanwair** [eslint/eslint#13514]
Eslint v: v7.5.0

**What I did:**

I got class defining inisideFunc(). I get this code, exmp:
```
export class Hello {
  insideFunc() { ... }
  render() { ... }
}
```
but I require to use only arrow functions instead normal function, except render, exmp:
```
export class Hello {
  insideFunc = () => { ... }
  render() { ... }
}
```

**What I did**
I already tried `func-style: "error"` or third party library like  `eslint-plugin-prefer-arrow` but it didn't show any error, exmp:

```
func-style: "error"

    "prefer-arrow/prefer-arrow-functions": [
     "error",
     {
       "disallowPrototype": true,
       "singleReturnOnly": false,
       "classPropertiesAllowed": false
     }
   ]
```

**What I expect**

I would like to use only arrow functions in my React component

**Comment by mdjermanovic** [comment]
Hi @spanwair, thanks for the issue!

Class fields are still in [stage 3](https://github.com/tc39/proposal-class-fields) and per our [policy about experimental features](https://github.com/eslint/eslint#what-about-experimental-features) the core rules support only stage 4 syntax.

We can evaluate this proposal if and when this syntax reaches stage 4, but since this rule seems to be a React-specific rule, it would be a better fit for [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react).

**Comment by mdjermanovic** [comment]
I'm closing this issue as it isn't currently actionable. Also, this looks like a rule for the React plugin rather than the core set of rules.
